src/shader/camera.rs

```diff
@@ -1,14 +1,34 @@
 use std::f32::consts::PI;
-
 use vek::Mat4;
-
+use vek::Vec4;
+use serde::{Serialize, Deserialize};
 use crate::config;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(default)]
+#[serde(deny_unknown_fields)]
 pub struct Camera {
-    //TODO(#4): These fields should be replaced with: position, target, fov, aspect_ratio_x, etc.
-    pub(crate) view: Mat4<f32>,
-    pub(crate) projection: Mat4<f32>,
+    pub(crate) eye: Vec4<f32>,
+    pub(crate) target: Vec4<f32>,
+    pub(crate) fovy: f32,
+    pub(crate) aspect_ratio_x: f32,
+    pub(crate) aspect_ratio_y: f32,
+    pub(crate) near: f32,
+    pub(crate) far: f32,
```

Comment:
Are any of these fields used in places other than this module? If not, `pub(crate)` can be removed from all of them. Struct fields are always visible within the same module, they only need to be public if they're being used in other modules/files. :)